Architecture_Engine/Query/IsValidForDistanceCheck.cs

```diff
@@ -0,0 +1,100 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Architecture.BuildersWork;
+using BH.oM.Architecture.Elements;
+using BH.oM.Geometry;
+using System.Collections.Generic;
+using System.Linq;
+using BH.oM.Spatial.ShapeProfiles;
+using BH.Engine.Geometry;
+using BH.oM.Geometry.CoordinateSystem;
+using System;
+
+namespace BH.Engine.Architecture
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        public static bool IsValidForDistanceCheck(this Opening opening1, Opening opening2, double maxDistance = double.NaN, double distanceTolerance = Tolerance.Distance, double angleTolerance = Tolerance.Angle)
+        {
+            // Check if the openings are parallel
+            if (opening1.CoordinateSystem.Z.IsParallel(opening2.CoordinateSystem.Z, angleTolerance) == 0)
```

Comment:
I haven't checked but why is `IsParallel` returning a number? Shouldn't it be returning a `true`/`false`? @pawelbaran maybe you know about this? Just seems odd to me that we would need to `== 0`?

Equally, if we must do `== 0` because `IsParallel` is being weird and returning a number rather than a `bool`, then it is probably good to add a tolerance, in case the return is `0.00000000000000000001` which is `!= 0` so checking for a range of acceptable values on a tight tolerance is probably preferable?

Reply:
This method returns an `int` as follows:
- -1 when the vectors have same direction, but different turn
- 0 when the vectors are not parallel
- 1 when the vectors have same direction and turn 

It would be faster if you looked up the method instead of writing lengthy comments @FraserGreenroyd 🙈 😉